### PR TITLE
added base class for various actvities

### PR DESCRIPTION
**Pull request by sanyud24**
cleaned the existing tests to inherit from the base test to follow DRY principle in the testing code.

Screenshot from 2022-06-13 19-47-44

### CODE REVIEW

**Comment by sanyud24** [comment]
Hi @Lamouresparus @wbrunette , here are the updates for (Outreachy) week2.

File: services_app/src/androidTest/java/org/opendatakit/activites/LoginActivity/BaseLoginActivity.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    public void clearTestEnvironment() {
+        activityScenario.onActivity(activity -> {
+            PropertiesSingleton props = activity.getProps();
+            assertThat(props).isNotNull();
+
+            Map<String, String> serverProperties = UpdateServerSettingsFragment.getUpdateUrlProperties(
+                    activity.getString(org.opendatakit.androidlibrary.R.string.default_sync_server_url)
+            );
+            assertThat(serverProperties).isNotNull();
+            serverProperties.put(CommonToolProperties.KEY_FIRST_LAUNCH,"true");
```
Comment (wbrunette) [inline services_app/src/androidTest/java/org/opendatakit/activites/LoginActivity/BaseLoginActivity.java:60]: I am not sure why KEY_FIRST_LAUNCH set to true should be a clean for the test environment. We can discuss in our meeting.

**Comment by sanyud24** [comment]
Hi @wbrunette, did the required changes by setting up `KEY_FIRST_LAUNCH = false` for the default setup function and added a separate test when `KEY_FIRST_LAUNCH = true,` which opens a dialog box that configures server settings.

**Comment by sanyud24** [comment]
Hi @wbrunette, did the required changes 

- setting up `KEY_FIRST_LAUNCH = false` for the default setup function and added a separate test when `KEY_FIRST_LAUNCH = true`, which opens a dialog box that configures server settings for the **MainActivity.**

- As KEY_FIRST_LAUNCH is set to false for **AbsSyncBaseActivity** `props.setProperties(Collections.singletonMap(CommonToolProperties.KEY_FIRST_LAUNCH, Boolean.toString(false)));` , so removed the reductant code and updated the tests with `KEY_FIRST_LAUNCH = false`

- classes extending from AbsSyncBaseActivity : LoginActivity, SyncActivity, VerifyServerSettingActivity

**Pull request merged**